completed programs extracted from !11

edwin1729 opened the pull request:
Those programs with no open review threads or assumes in PR 11 are copied over here to be merged

The following programs were dropped
007
037
072
077

see original commits and review at https://github.com/secure-foundations/human-eval-verus/pull/11

coauthored by @WeetHet

edwin1729 commented:
There seems to be a problem with the CI. The required version of rustc isn't being installed in the vm

parno commented:
Thanks, it will be great to get these added in.  I updated the CI script on `main`, so if you merge that in, hopefully that will fix the CI on your branch.

Looking back at #11, it seems like 037 might be complete.  Were there still pending issues with it?

edwin1729 commented:
> Looking back at #11, it seems like 037 might be complete. Were there still pending issues with it?

There were over 5 programs which I believe used to compile but broke with an error along the lines of while loop requires a decreases clause. I have fixed all of those except 037. 037 gives further errors on fixing the while error which I don't quite understand so I dropped it.

edwin1729 commented:
also ci is now failing due to merged programs being broken by backwards incompatible changes to verus

parno commented:
> also ci is now failing due to merged programs being broken by backwards incompatible changes to verus

Hmm, indeed.  I'll try to fix that up early next week and then get this merged.

parno commented:
Okay, I fixed up the existing files, so CI passes on `main`.  I manually checked your modified files, and they pass, so I'll merge this in.
